### ISSUE TITLE
image-level pooling

### ISSUE THREAD

**Issue by edgarmedina1801** [bonlime/keras-deeplab-v3-plus#41]
Hello,

I checked your code, I think the image-level pooling was not correctly implemented. because it is not computing the global average pooling.

https://github.com/bonlime/keras-deeplab-v3-plus/blob/master/model.py#L437

I checked this code:
https://github.com/tensorflow/models/blob/4a0ee4a29dd7e4b6e0135ccf6857f4dc58d71a90/research/deeplab/model.py#L397
and Reference are here ref 52 from original paper (https://arxiv.org/pdf/1506.04579.pdf):
> Exploiting the FCN architecture, ParsetNet can directly use global average pooling from the final (or any) feature map, resulting in the feature of the whole
image, and use it as context.
> Specifically, we use global average pooling and pool the context features from the last layer or any
layer if that is desired.

**Comment by bonlime** [comment]
Hi,
Can you suggest a way to modify the code, so that feature pooling would be implemented correctly?

**Comment by edgarmedina1801** [comment]
I'm trying to implement this network in Pytorch, so, maybe in keras this works:

x = GlobalAveragePooling2D()(x)
...
x = BilinearUpsampling((input_shape[0],input_shape[1]))(b4)

If you have problems with convolutions due to Nx1x1xNum_features, maybe a Linear layer can help. because 1x1 conv in a Nx1x1xNum_features volume is the same as a linear layer.

Thanks for fixing the bug.

**Comment by bhack** [comment]
The crop case is set in https://github.com/tensorflow/models/blob/master/research/deeplab/train.py#L198

**Comment by edgarmedina1801** [comment]
Oh I see, I didn't see that part. Thank you for sharing and for clearing that up. I also saw it supports both methods.

**Comment by bhack** [comment]
Yes https://github.com/tensorflow/models/blob/master/research/deeplab/model.py#L394
